### PR TITLE
[Assignment 2] Adding Assignment endpoints and CRUD functionality

### PR DESCRIPTION
**Pull request by andrewpaulino**
I need to add some functionality to ensure our gradebook backend can handle adding assignments.

I added 3 endpoints
* One to create an assignment
* One to delete an assignment
* One to modify the title of an assignment
 
I also added JUnit test to ensure the endpoint is calling the database and returning 200.

### CODE REVIEW

File: cst438_gradebook/src/main/java/com/cst438/domain/Assignment.java [hunk 1]
```diff
@@ -54,6 +62,8 @@ public void setNeedsGrading(int needsGrading) {
 		this.needsGrading = needsGrading;
 	}
 	
+
+
```
Comment (lyratadev) [inline cst438_gradebook/src/main/java/com/cst438/domain/Assignment.java:66]: Maybe fix the whitespace so we don't pollute the changelog.

File: cst438_gradebook/src/main/java/com/cst438/domain/AssignmentGradeRepository.java [hunk 2]
```diff
@@ -11,4 +11,4 @@ AssignmentGrade findByAssignmentIdAndStudentEmail(
 			@Param("assignmentId") int assignmentId, 
 			@Param("email") String email );
 	
-}
+}
```
Comment (lyratadev) [inline cst438_gradebook/src/main/java/com/cst438/domain/AssignmentGradeRepository.java:14]: Also here change the whitespace so we don't pollute the changelog.

File: cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java [hunk 3]
```diff
@@ -243,6 +244,116 @@ public void updateAssignmentGrade() throws Exception {
 		verify(assignmentGradeRepository, times(1)).save(updatedag);
 	}
 
+	public void assignmentCrudOperations() throws Exception {
```
Comment (lyratadev) [inline cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java:247]: Make sure you add @Test so Junit can pick up this as a test.

File: cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java [hunk 4]
```diff
@@ -10,4 +10,7 @@ public interface AssignmentRepository extends CrudRepository <Assignment, Intege
 
 	@Query("select a from Assignment a where a.needsGrading=1 and a.dueDate < current_date and a.course.instructor= :email order by a.id")
 	List<Assignment> findNeedGradingByEmail(@Param("email") String email);
+	
+	@Query("select a from Assignment a where a.needsGrading=0 and a.dueDate < current_date and a.id = :assignmentId order by a.id")
```
Comment (lyratadev) [inline cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java:14]: Looks good!

**Review by lyratadev — APPROVED** [review]

File: cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java (hunk 3 above)
Comment (andrewpaulino) [reply cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java:247]: Noted!

File: cst438_gradebook/src/main/java/com/cst438/domain/Assignment.java (hunk 1 above)
Comment (andrewpaulino) [reply cst438_gradebook/src/main/java/com/cst438/domain/Assignment.java]: Good call!

File: cst438_gradebook/src/main/java/com/cst438/domain/AssignmentGradeRepository.java (hunk 2 above)
Comment (andrewpaulino) [reply cst438_gradebook/src/main/java/com/cst438/domain/AssignmentGradeRepository.java]: Good call!

**Pull request merged**